vertical-pod-autoscaler/pkg/admission-controller/config.go

```diff
@@ -89,6 +89,22 @@ func selfRegistration(clientset *kubernetes.Clientset, caCert []byte, namespace,
 							Resources:   []string{"verticalpodautoscalers"},
 						},
 					},
+					{
+						Operations: []admissionregistration.OperationType{admissionregistration.Create, admissionregistration.Update},
+						Rule: admissionregistration.Rule{
+							APIGroups:   []string{"apps"},
+							APIVersions: []string{"v1"},
+							Resources:   []string{"deployments"},
+						},
+					},
+					{
+						Operations: []admissionregistration.OperationType{admissionregistration.Create, admissionregistration.Update},
```

Comment:
Same remark about `update` , except that here we don't even have the ReplicaSet option.
That may lead us to diff old spec vs new spec to see if the change would have triggered a rolling update of the pods.